### PR TITLE
Create tests

### PR DESCRIPTION
**Pull request by ItaloG**
Description
This pull request implements tests on usecases, presentations, utils and other functions. I've refactor some inpropert dependencies injection and add TODO comments to codes i'm not aware of.

What type of PR contains?
- [ ] 🍕 Feature
- [ ]  🐛 Bug Fix
- [ ]  📝 Documentation Update
- [ ]  🔥 Performance Improvements
- [X]  🧑‍💻 Code Refactor
- [X]  ✅ Test
- [ ]  🤖 Build
- [ ]  🔁 CI
- [ ]  📦 Chore (Release)
- [ ]  ⏩ Revert

### CODE REVIEW

**Comment by lucassm02** [comment]
@ItaloG , please resolve the conflicts so that I can proceed with the review.

**Comment by ItaloG** [comment]
@lucassm02 , I've resolved the conflicts.

File: test/integration/infra/worker/worker-manager.spec.ts
Comment (lucassm02) [inline test/integration/infra/worker/worker-manager.spec.ts:1]: If the implementation is not correct, please remove the file

File: src/util/object/formatters/recursive-data-converter.ts [hunk 1]
```diff
@@ -2,22 +2,26 @@ export const recursiveDataConvertFilterLayer = (
...
     return recursiveDataConvertApplyLayer(data, formatter);
   return data;
 };
+
 const recursiveDataConvertApplyLayer = (
   object: Record<string, unknown>,
   formatter: (params: any) => any
 ): Record<string, unknown> => {
+  // TODO: I don't know how to get this case
   if (!object) return object;
```
Comment (lucassm02) [inline src/util/object/formatters/recursive-data-converter.ts:25]: This is just a validation at the implementation level; null or undefined should be passed as arguments.

File: src/util/object/formatters/recursive-data-converter.ts [hunk 2]
```diff
@@ -2,22 +2,26 @@ export const recursiveDataConvertFilterLayer = (
   data: any,
   formatter: (params: any) => any
 ) => {
-  if (Array.isArray(data))
+  if (Array.isArray(data)) {
     return data.map((item) => {
+      // TODO: this condition is correct? This should get instanceof item or data?
```
Comment (lucassm02) [inline src/util/object/formatters/recursive-data-converter.ts:7]: The validation is correct; it's to prevent objects of type `Date` from being modified

**Review by lucassm02 — CHANGES_REQUESTED** [review]
I hope I've answered some of your questions. Thank you for adding those `TODOs`; it contributes a lot to the project's quality. I'll address them soon. Please just verify my request for file deletion.

**Comment by ItaloG** [comment]
The changes have been added

**Review by lucassm02 — APPROVED** [review]
@ItaloG thank you for the implementation. It's of great importance to the project, and it will greatly assist with the new implementations!

**Pull request merged**